winch/codegen/src/codegen/mod.rs

```diff
@@ -166,6 +174,9 @@ where
                 _ => {}
             },
         );
+        if masm.sp_offset() > target_sp {
+            masm.free_stack(masm.sp_offset() - target_sp);
+        }
```

Comment:
Can we also assert that `masm.sp_offset() - target_sp` equals the size of all the `Memory` entries in the value stack that got dropped?